### PR TITLE
Cleanup codec constants

### PR DESCRIPTION
**Pull request by abi87**
## Why this should be merged
This PR used to bring some changes to the codec that are not necessary anymore given the way I restructured dynamic fees.
Instead of fully removing the PR, I kept only some minor nits.

## How this works
Just nits

## How this was tested
CI

### CODE REVIEW

File: codec/manager.go [hunk 1]
```diff
@@ -13,6 +13,8 @@ import (
 )
 
 const (
+	CodecVersionSize = wrappers.ShortLen
```
Comment (StephenButtolph) [inline codec/manager.go:16]: I think this stutters:
```suggestion
	VersionSize = wrappers.ShortLen
```

File: codec/reflectcodec/type_codec.go [hunk 2]
```diff
@@ -502,7 +496,7 @@ func (c *genericCodec) marshal(
 // interface
 func (c *genericCodec) Unmarshal(bytes []byte, dest interface{}) error {
 	if dest == nil {
-		return errUnmarshalNil
+		return codec.ErrUnmarshalNil
```
Comment (StephenButtolph) [inline codec/reflectcodec/type_codec.go:499]: nice

File: codec/reflectcodec/type_codec.go [hunk 3]
```diff
@@ -84,7 +82,7 @@ func New(typer TypeCodec, tagNames []string) codec.Codec {
 
 func (c *genericCodec) Size(value interface{}) (int, error) {
 	if value == nil {
-		return 0, errMarshalNil // can't marshal nil
+		return 0, nil // can't marshal nil, we return zero size
```
Comment (StephenButtolph) [inline codec/reflectcodec/type_codec.go:85]: I'm not exactly sure this is needed (talked offline)

**Review by StephenButtolph — APPROVED** [review]

File: codec/manager.go [hunk 4]
```diff
@@ -13,6 +13,8 @@ import (
 )
 
 const (
+	VersionSize = wrappers.ShortLen
```
Comment (dhrubabasu) [inline codec/manager.go:16]: do we want to do VersionLen?

**Review by dhrubabasu — APPROVED** [review]

File: codec/manager.go (hunk 4 above)
Comment (StephenButtolph) [reply codec/manager.go:16]: I thought about this... I think it makes sense as `Size` because it's used in the `Size` method... 🤷

**Pull request merged**